tgui/packages/tgui/interfaces/ModularShieldGen.tsx

```diff
@@ -0,0 +1,147 @@
+import { useBackend } from '../backend';
+import { Window } from '../layouts';
+import { Stack, Section, ProgressBar, Button, NumberInput } from '../components';
+
+type modularShieldGeneratorData = {
+  max_strength: number;
+  current_strength: number;
+  max_regeneration: number;
+  current_regeneration: number;
+  max_radius: number;
+  current_radius: number;
+  active: Boolean;
+  recovering: Boolean;
+  exterior_only: Boolean;
```

Comment:
BooleanLike

Reply:
"cannot find name BooleanLike" errors

Reply:
need to import BooleanLike. you should be able to hover over it / get a blue light bulb icon

Reply:
![Screenshot 2023-05-17 152710](https://github.com/tgstation/tgstation/assets/42397676/64fb2520-1439-43c4-a5ae-76c67d10c17b)

Reply:
thanks, i dont get lightbulbs, i looked at the autolathe interface code and saw how it imports BooleanLike and used that.

Reply:
which editor are you using?

Reply:
visual studio code